src/main/scala/PSO/Particle.scala

```diff
@@ -70,42 +78,49 @@ class Particle(initVelocity: List[Double], inertia: Double, localAccel: Double,
     // Print local and neighbourhood bests to console
     case ReportRequest =>
       sender ! Report(currVel, currPos, localBest, neighbourhoodBest)
-      // println(s"Particle: lbest = $localBest, nbest = $neighbourhoodBest")
     // Handle unknown messages
     case _ =>
       throw new RuntimeException("Unknown particle message type")
   }
 
-  private def computeIteration() = {
+  def computeIteration(): Unit = {
+    iter += 1
```

Comment:
Should the particle know about the termination criteria (iteration count)? I feel like thats the job of the master, and then when it's met send a PoisonPill to the actors.